### PR TITLE
Dataaccess

### PR DESCRIPTION
**Pull request by 2end**
Patterns: Repository, UnitOfWork

### CODE REVIEW

**Comment by VitalikKoval** [comment]
agree

File: RAS.DataAccess/Implementation/RASContext.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            base.OnModelCreating(modelBuilder);
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=repositorydb;Trusted_Connection=True;");
```
Comment (Yuriy-Pelekh) [inline RAS.DataAccess/Implementation/RASContext.cs]: Extract connection string into config file. Does not commit real credentials!

File: RAS.DataAccess/Implementation/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore;
+using RAS.DataAccess.Interfaces;
+
+namespace RAS.DataAccess.Implementation
+{
+    public class Repository<TEntity> : IRepository<TEntity> where TEntity : class
```
Comment (Yuriy-Pelekh) [inline RAS.DataAccess/Implementation/Repository.cs]: Please write `where` clause in another line like this:
```
public class Repository<TEntity> : IRepository<TEntity>
       where TEntity : class

```

File: RAS.DataAccess/Implementation/UnitOfWork.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            return _context.SaveChanges();
+        }
+
+        public void Dispose()
+        {
+            _context.Dispose();
+        }
+
+        public IRepository<TEntity> GetRepository<TEntity>() where TEntity : class
```
Comment (Yuriy-Pelekh) [inline RAS.DataAccess/Implementation/UnitOfWork.cs]: Same as above.

File: RAS.DataAccess/Interfaces/IRepository.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace RAS.DataAccess.Interfaces
+{
+    public interface IRepository<TEntity> where TEntity : class
```
Comment (Yuriy-Pelekh) [inline RAS.DataAccess/Interfaces/IRepository.cs]: Same as above.

File: RAS.DataAccess/Interfaces/IRepository.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        TEntity Get(int id);
+
+        // Returns all entities
```
Comment (Yuriy-Pelekh) [inline RAS.DataAccess/Interfaces/IRepository.cs]: Redundant comment.

File: RAS.DataAccess/Interfaces/IUnitOfWork.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace RAS.DataAccess.Interfaces
+{
+    public interface IUnitOfWork : IDisposable
+    {
+        IRepository<TEntity> GetRepository<TEntity>() where TEntity : class;
```
Comment (Yuriy-Pelekh) [inline RAS.DataAccess/Interfaces/IUnitOfWork.cs]: Same as above.

File: RAS.DataAccess/Models/Group.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        public DateTime EndDate { get; set; }
+        public int Free { get; set; }
+        public int HasEng { get; set; }
+        public int HasFirst { get; set; }
+        public int HasTech { get; set; }
+        public string Name { get; set; }
+        public int NotSynchronized { get; set; }
+        public DateTime StartDate { get; set; }
+        public int Status { get; set; }
+
```
Comment (Yuriy-Pelekh) [inline RAS.DataAccess/Models/Group.cs]: Redundant empty line. Please be consistent.

**Pull request merged**